lib/ansible/modules/async_wrapper.py

```diff
@@ -74,7 +74,7 @@ def daemonize_self():
 
 # NB: this function copied from module_utils/json_utils.py. Ensure any changes are propagated there.
 # FUTURE: AnsibleModule-ify this module so it's Ansiballz-compatible and can use the module_utils copy of this function.
-def _filter_non_json_lines(data):
+def _filter_non_json_lines(data: str) -> str:
```

Comment:
The last line is `return ('\n'.join(lines), warnings)` which is a tuple of a string and a list of strings (which MyPy rightfully complains about):
```suggestion
def _filter_non_json_lines(data: str) -> tuple[str, list[str]]:
```